tests/Microsoft.Identity.Test.Integration.net45/HeadlessTests/OboTests.cs

```diff
@@ -65,6 +70,60 @@ public async Task OBO_WithCache_MultipleUsers_Async()
             await RunOnBehalfOfTestAsync(aadUser2, true).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        [Timeout(2 * 60 * 1000)] // 2 min timeout
```

Comment:
>[Timeout(2 * 60 * 1000)] // 2 min timeout [](start = 8, length = 41)

any reason there has to be this timeout?

Reply:
it's doing device code flow, so that's why the time out is there. i can think about a different way of doing this.

---
In reply to: [345533947](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1496#discussion_r345533947) [](ancestors = 345533947)

Reply:
There are several levels of timeout: 

1. Each Selenium operation has an implicit timeout (I believe 10 sec). So if clicking on a button takes more than 10s -> fail. 
2. Some Selenium operations have custom timeouts (e.g. wait for a button to be enabled for 20 seconds)
3. TcpListener has I believe a 5 min timeout, in case smth happens to the browser / redirect etc.

However, the test can hang outside Selenium and the tcp listener - bugs in the tcp listener / Selenioum syncronization, bugs in MSAL etc. This happens rarely, but when it happens I'd rather not have to wait 1h for the entire build to timeout.